src/executables/master.c

```diff
@@ -441,19 +465,23 @@ static void
 css_register_new_server (CSS_CONN_ENTRY * conn, unsigned short rid)
 {
   int name_length;
-  char *server_name = NULL;
+  std::string server_name;
+  SERVER_TYPE type;
   SOCKET_QUEUE_ENTRY *entry;
 
   /* read server name */
-  if (css_receive_data (conn, rid, &server_name, &name_length, -1) == NO_ERRORS)
+  if (receive_server_info (conn, rid, server_name, type) == NO_ERRORS)
     {
-      entry = css_return_entry_of_server (server_name, css_Master_socket_anchor);
+      entry = css_return_entry_of_server (server_name.c_str (), css_Master_socket_anchor);
+      name_length = server_name.length ();
+
       if (entry != NULL)
 	{
+	  entry->server_type = type;
 	  if (IS_INVALID_SOCKET (entry->fd))
 	    {
 	      /* accept a server that was auto-started */
-	      css_accept_old_request (conn, rid, entry, server_name, name_length);
+	      css_accept_old_request (conn, rid, entry, server_name.c_str (), name_length);
```

Comment:
Remove name_length variable. Same for css_register_new_server2.
```suggestion
	      css_accept_old_request (conn, rid, entry, server_name.c_str (), (int) server_name.length ());
```